server/src/main/java/org/elasticsearch/common/time/Iso8601Parser.java

```diff
@@ -462,6 +462,9 @@ private ZoneId parseZoneId(CharSequence str, int pos) {
             }
         }
         pos++;  // read the + or -
+        if (str.charAt(pos) == '[' && str.charAt(len - 1) == ']') {
```

Comment:
I think this could lead to some malformed strings being parsed, like `2022-07-08T00:14:07+[Europe/London]`